src/platform/plugins/shared/share/public/components/export_integrations/export_integrations.test.tsx

```diff
@@ -89,6 +89,66 @@ describe('Export Integrations', () => {
     });
   });
 
+  describe('Export Derivatives', () => {
+    const exportDerivativeLabel = 'Export derivative';
+    const getMockExportDerivativeConfig = (shouldRender: boolean): ShareConfigs => ({
+      shareType: 'integration',
+      groupId: 'exportDerivatives',
+      id: 'anExampleExportDerivative',
+      config: {
+        shouldRender: () => shouldRender,
+        label: () => <span>{exportDerivativeLabel}</span>,
+        flyoutContent: () => <div />,
+        icon: 'empty',
+      },
+    });
+
+    it('render export derivatives with passing shouldRender predicates', async () => {
+      const user = userEvent.setup();
+
+      render(
+        <ExportPopoverRender
+          shareContext={{
+            ...mockShareContext,
+            shareMenuItems: [
+              ...mockShareContext.shareMenuItems,
+              getMockExportDerivativeConfig(true),
+            ],
+          }}
+        />
+      );
+
+      await user.click(screen.getByText('click me'));
+
+      await waitForEuiPopoverOpen();
+
+      expect(screen.getByText(exportDerivativeLabel)).toBeInTheDocument();
+    });
+
+    it('does not render export derivatives with non-passing shouldRender predicates', async () => {
+      const user = userEvent.setup();
+      const label = 'Export derivative';
+
+      render(
+        <ExportPopoverRender
+          shareContext={{
+            ...mockShareContext,
+            shareMenuItems: [
+              ...mockShareContext.shareMenuItems,
+              getMockExportDerivativeConfig(false),
+            ],
+          }}
+        />
+      );
+
+      await user.click(screen.getByText('click me'));
+
+      await waitForEuiPopoverOpen();
+
+      expect(screen.queryByText(label)).not.toBeInTheDocument();
```

Comment:
It might be easier to reason about this test, if the same identifier that was provided as the label for the returned property in `getMockExportDerivativeConfig` is what is been validated

Reply:
My bad, that's what I wanted to do when I moved the label up the scope 🙂